genestack_client/scripts/genestack_application_manager.py

```diff
@@ -536,6 +533,15 @@ def upload_single_file(application, file_path, version, override,
     if not no_wait:
         for app_id in app_info.identifiers:
             wait_application_loading(application, app_id, version)
+        if app_info.identifiers:
+            # Get jar loading messages from one of its applications:
+            descriptor = application.invoke('getApplicationDescriptor', app_info.identifiers[0], version)
```

Comment:
I tried to implement waiting function with callback and this solution looks to complex and difficult for usage. So, i decided to simply return tuple with success flag and descriptor from `wait_application_loading` function.